src/sql/parts/connection/connectionDialog/connectionDialogService.ts

```diff
@@ -326,7 +333,10 @@ export class ConnectionDialogService implements IConnectionDialogService {
 			let container = withElementById(this._partService.getWorkbenchElementId()).getHTMLElement().parentElement;
 			this._container = container;
 			this._connectionDialog = this._instantiationService.createInstance(ConnectionDialogWidget, this._providerTypes, this._providerNameToDisplayNameMap[this._model.providerName]);
-			this._connectionDialog.onCancel(() => this.handleOnCancel(this._connectionDialog.newConnectionParams));
+			this._connectionDialog.onCancel(() => {
+				this._connectionDialog.databaseDropdownExpanded = this.uiController.databaseDropdownExpanded;
```

Comment:
Not sure this logic belongs in the cancellation handler. Shouldn't the connection dialog's `databaseDropdownExpanded` property always match the controller's?